example/src/main/java/andrewoid/neuralnetwork/MNISTReader/MNISTTraining.java

```diff
@@ -0,0 +1,72 @@
+package andrewoid.neuralnetwork.MNISTReader;
+
+import andrewoid.neutralnetwork.Network;
+import andrewoid.neutralnetwork.Neuron;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Random;
+
+public class MNISTTraining
+{
+    private static final String trainingInputImagePath = "train-images-idx3-ubyte";
+    private static final String trainingInputLabelPath = "train-labels-idx1-ubyte";
+    private static final String testingInputImagePath = "t10k-images-idx3-ubyte";
+    private static final String testingInputLabelPath = "t10k-labels-idx1-ubyte";
+
+    private static final String trainingOutputPath = "TrainingMNISTOutputFiles/";
+    private static final String testingOutputPath = "TestingMNISTOutputFiles/";
+
+    private static final double expectedOutputs[][] = new double[][]{
+            {1, 0, 0, 0, 0, 0, 0, 0, 0, 0}, // 0
+            {0, 1, 0, 0, 0, 0, 0, 0, 0, 0}, // 1
+            {0, 0, 1, 0, 0, 0, 0, 0, 0, 0}, // 2
+            {0, 0, 0, 1, 0, 0, 0, 0, 0, 0}, // 3
+            {0, 0, 0, 0, 1, 0, 0, 0, 0, 0}, // 4
+            {0, 0, 0, 0, 0, 1, 0, 0, 0, 0}, // 5
+            {0, 0, 0, 0, 0, 0, 1, 0, 0, 0}, // 6
+            {0, 0, 0, 0, 0, 0, 0, 1, 0, 0}, // 7
+            {0, 0, 0, 0, 0, 0, 0, 0, 1, 0}, // 8
+            {0, 0, 0, 0, 0, 0, 0, 0, 0, 1}, // 9
+    };
+
+    public static void main(String args[])
+    {
+        IdxReader reader = new IdxReader(trainingInputImagePath,trainingInputLabelPath,trainingOutputPath);
+        if(reader.areFilesLoaded())
```

Comment:
You shouldn't need to call `areFilesLoaded()` 
You do not need the uncompressed files to do this.

Reply:
Good point. Will resolve